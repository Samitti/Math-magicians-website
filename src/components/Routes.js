import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import Home from './Home';
import Quote from './Quote';
import Navbar from './Navbar';
import App from './App';

const Routes = () => (
  <BrowserRouter>
    <Navbar />
    <Switch>
      <Route exact path="/" component={Home} />
      <Route exact path="/App" component={App} />
      <Route exact path="/Quote" component={Quote} />
    </Switch>
  </BrowserRouter>
);

export default Routes;
