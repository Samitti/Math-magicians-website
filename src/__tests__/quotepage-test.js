import React from 'react';
import ReactDOM from 'react-dom';
import Routes from '../components/Routes';

it('checks the snapshot of quotepage', () => {
    const div = document.createElement('div')
    document.body.appendChild(div)
    ReactDOM.render(<Routes />, div)
    const quuotepage = document.body.querySelector('#quotepage')
    quotepage.click()
    expect(div).toMatchSnapshot();
});
