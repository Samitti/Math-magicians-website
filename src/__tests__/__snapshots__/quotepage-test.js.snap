// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks the snapshot of quotepage 1`] = `
<div>
  <div
    class="topAll"
  >
    <h1>
      Math Magicians
    </h1>
    <nav
      class="navLink"
    >
      <p>
        <a
          href="/"
          id="homepage"
          style="color: green;"
        >
          Home
        </a>
      </p>
      <p
        class="linkSplit"
      >
        |
      </p>
      <p>
        <a
          href="/App"
          id="calculatorpage"
          style="color: green;"
        >
          Calculator
        </a>
      </p>
      <p
        class="linkSplit"
      >
        |
      </p>
      <p>
        <a
          href="/Quote"
          id="quotepage"
          style="color: green;"
        >
          Quote
        </a>
      </p>
    </nav>
  </div>
  <div
    class="quoteContainer"
  >
    <h2>
      Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. â€“William Paul Thurston
    </h2>
  </div>
</div>
`;
